File: nomad-samples/python/basic-live-channel-1.1/schedule_event/add_input_schedule_event.py
from constants.system_constants import *
from constants.project_constants import *
from exceptions.api_exception_handler import *

from libraries import json, requests

'''
 * Add Input Schedule Event
 *
 * @param {string} AUTH_TOKEN    | Authentication token
 * @param {Object} DATA         | BODY
 *
 * @returns JSON Object
'''
async def add_input_schedule_event(AUTH_TOKEN, DATA):
    # Check for valid parameters
    if (not AUTH_TOKEN or not DATA):
        raise Exception("Add Input Schedule Event: Invalid API call")


    # Build the payload BODY
    BODY = {
        "channelId": DATA["channelId"],
        "type": {
            "id": LIVE_INPUT_LOOKUP_ID,
            "description": "Live Input"
        },
        "liveInput": {
            "id": DATA["inputId"],
            "description": "name"#DATA["name"]
        },
        "previousId": DATA["previousId"]
    }


    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        # Send the request
        RESPONSE = requests.post(SERVER_URL + "/liveChannel/" + DATA["channelId"] + "/liveScheduleEvent",  headers= HEADERS, data= json.dumps(BODY))
    
        # Return JSON response
        INFO =  json.loads(RESPONSE.text)

        if not RESPONSE.ok:
            raise Exception()
        
        return INFO

    except:
        await api_exception_handler(RESPONSE, "Add Input Schedule Event failed")

File: nomad-samples/admin/js/basic-live-channel/schedule-event/add-input-schedule-event.js
import * as sysConstants from "../constants/system-constants.js";
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * Add Input Schedule Event
 *
 * @param {string} authToken    | Authorization Token
 * @param {Object} data         | Body
 *
 * @returns JSON Object
 */
export default async function addInputScheduleEvent(authToken, data) {
    // Check for valid parameters
    if (!authToken || !data) {
        throw new Error("Add Input Schedule Event: Invalid API call");
    }

    // Build the payload body
    const BODY = {
        channelId: data.channelId,
        type: {
            lookupId: sysConstants.LIVE_INPUT_LOOKUP_ID,
            description: "Live Input"
        },
        liveInput: {
            lookupId: data.inputId,
            description: data.name
        },
        previousId: data.previousId
    };

    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const response = await fetch(`${prjConstants.SERVER_URL}/LiveChannel/${data.channelId}/liveScheduleEvent`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    });

    // Check for success
    if (response && response.ok) {
        // Return JSON response
        return await response.json();
    }

    await apiExceptionHandler(response, "Add Input Schedule Event failed");
}
